XcodeAutoBasher/XcodeAutoBasher/Models/VOKProjectContainer.m

```diff
@@ -116,6 +116,52 @@ - (VOKDirectoryWatcher *)directoryWatcher
     return _directoryWatcher;
 }
 
+- (NSDictionary *)environmentVariables
+{
+    if ([NSThread isMainThread]) {
+        NSLog(@"VOKProjectContainer environmentVariables being retrieved on the main thread.");
+    }
+    NSPipe *outputPipe = [NSPipe pipe];
+    
+    NSTask *buildSettingsTask = [[NSTask alloc] init];
+    buildSettingsTask.launchPath = @"/usr/bin/xcodebuild";
+    buildSettingsTask.currentDirectoryPath = self.containingPath;
+    buildSettingsTask.arguments = @[ @"-showBuildSettings", ];
+    buildSettingsTask.standardOutput = outputPipe;
+    
+    [buildSettingsTask launch];
+    [buildSettingsTask waitUntilExit];
+    
+    NSData *outputData = [outputPipe.fileHandleForReading readDataToEndOfFile];
+    NSString *outputString = [[NSString alloc] initWithData:outputData encoding:NSUTF8StringEncoding];
+    
+    NSError *error;
+    NSRegularExpression *regex = [NSRegularExpression
```

Comment:
Regexes are not pure voodoo.  That said, how much detail?  Examples:
- `// Search for lines of the form [whitespace][key][whitespace]=[whitespace][value] and extract the key and value`
- http://stackoverflow.com/a/13827487/291280 and a note that this is an adaptation of that answer's regex
- http://rick.measham.id.au/paste/explain.pl?regex=%5E%5B%5B%3Aspace%3A%5D%5D*%28%5B%5E%5B%3Aspace%3A%5D%5D*%29%5B%5B%3Aspace%3A%5D%5D*%3D%5B%5B%3Aspace%3A%5D%5D*%28.*%29%24
- From https://regex101.com/:  
![image](https://cloud.githubusercontent.com/assets/7505072/8636739/15ab04a4-2837-11e5-9382-226430da3895.png)